tests/frontend/show_artifact_cas_digest.py

```diff
@@ -0,0 +1,291 @@
+
+        # Push the built artifacts to the remote cache
+        result = cli.run(project=project, args=["artifact", "push", target, "--artifact-remote", share.repo])
+        result.assert_success()
+
+        # Check the target has been built locally and is existing in the remote cache
+        assert cli.get_element_state(project, target) == "cached"
+        assert_shared(cli, share, project, target)
+
+        # Check the target has an artifact digest
+        result = cli.run(project=project, silent=True, args=["show", "--format", "%{name},%{artifact-cas-digest}", target])
+        result.assert_success()
+
+        received_digest = result.output.splitlines()[0]
+        assert received_digest == "{target},{digest}".format(target=target, digest=expected_digest)
+
+
+# This tests that a target that has been built locally and
+# that is cached remotely has an artifact CAS digest.
+#
+# The test is performed with a remote cache.
+#
+@pytest.mark.datafiles(DATA_DIR)
+@pytest.mark.parametrize(
+    "target",
+    [
+        ("import-basic-files.bst"),
+        ("import-executable-files.bst"),
+        ("import-symlinks.bst"),
+    ],
+)
+def test_show_artifact_cas_digest_cached_remotely_only(cli, tmpdir, datafiles, target):
+    project = str(datafiles)
+    expected_no_digest = ""
+
+    # Configure a local cache
+    local_cache = os.path.join(str(tmpdir), "cache")
+    cli.configure({"cachedir": local_cache})
+
+    with create_artifact_share(os.path.join(str(tmpdir), "artifactshare")) as share:
+
+        cli.configure({"artifacts": {"servers": [{"url": share.repo}]}})
+
+        # Build the target locally
+        result = cli.run(project=project, silent=True, args=["build", target])
+        result.assert_success()
+
+        # Push the built artifacts to the remote cache
+        result = cli.run(project=project, args=["artifact", "push", target, "--artifact-remote", share.repo])
+        result.assert_success()
+
+        # Delete the locally cached target
+        result = cli.run(project=project, silent=True, args=["artifact", "delete", target])
+        result.assert_success()
+
+        # Check the target has been deleted locally
+        assert cli.get_element_state(project, target) == "buildable"
+        assert_shared(cli, share, project, target)
+        
+        # Check the target has an artifact digest
+        result = cli.run(project=project, silent=True, args=["show", "--format", "%{name},%{artifact-cas-digest}", target])
+        result.assert_success()
+
+        received_digest = result.output.splitlines()[0]
+        assert received_digest == "{target},{digest}".format(target=target, digest=expected_no_digest)
+
+
+# This tests that the target and its dependencies are built
+# as expected and that all their artifact CAS digests are
+# available.
+#
+# The test is performed without a remote cache.
+#
+@pytest.mark.datafiles(DATA_DIR)
+@pytest.mark.parametrize(
+    "target, expected_digests",
```

Comment:
Same comment regarding using the `ids` parameter in parametrized tests which have complicated/long parameters.